astropy/units/physical.py

```diff
@@ -311,6 +313,19 @@ def __init__(self, unit, physical_types):
     def __iter__(self):
         yield from self._physical_type_list
 
+    def __getattr__(self, attr):
+        if attr not in dir(self) and attr in dir(str):
```

Comment:
According to https://docs.python.org/3/reference/datamodel.html#object.__getattr__.
We should never be in ``__getattr__`` if ``attr`` is in ``dir(self)``.